src/main/scala/org/allenai/scholar/metrics/ErrorAnalysis.scala

```diff
@@ -0,0 +1,41 @@
+package org.allenai.scholar.metrics
+
+import java.time.Year
+
+import org.allenai.scholar._
+import org.allenai.scholar.metrics.PrecisionRecall._
+
+case class ErrorAnalysis(
+  metricName: String,
+  pr: PR,
+  details: Iterable[(String, Example[_])]
+)
+
+object ErrorAnalysis {
+  def computeMetrics[T](
+    truth: Map[String, T],
+    predicted: Map[String, T],
+    metrics: (String, T => Iterable[_])*
+  ) = {
+
+    val examples: Iterable[(String, (String, Example[_]))] =
+      for {
+        (id, trueData) <- truth.toList
```

Comment:
This first loop produces one row each metric-paperId combination; there is
a groupBy on metric below.  But it is a good point that the `truth` input
should not be a Map.  I will change that in the next PR.